jpos/src/main/java/org/jpos/transaction/Context.java

```diff
@@ -298,7 +296,11 @@ protected void dumpEntry (PrintStream p, String indent, Map.Entry<Object,Object>
         Object value = entry.getValue();
         if (value instanceof Loggeable) {
             p.println("");
-            ((Loggeable) value).dump(p, indent + " ");
+            try {
+                ((Loggeable) value).dump(p, indent + " ");
+            } catch (Exception ex) {
+                ex.printStackTrace(p);
```

Comment:
Perhaps it would be nice to also print an explanation of what happened, so it is clearer at a first look to the std err, that it was an exception while dumping before having to pay attention to the stack trace.

This could even be a new log event with the message and the exception

Reply:
It's not printing to stderr, it goes to the PrintStream, along with the rest of the message.

Reply:
Last time I checked [ex.printStackTrace, printed to std err.](https://docs.oracle.com/javase/8/docs/api/java/lang/Throwable.html#printStackTrace--)

> Prints this throwable and its backtrace to the standard error stream. This method prints a stack trace for this Throwable object on the error output stream that is the value of the field System.err. The first line of output contains the result of the [toString()](https://docs.oracle.com/javase/8/docs/api/java/lang/Throwable.html#toString--) method for this object. Remaining lines represent data previously recorded by the method [fillInStackTrace()](https://docs.oracle.com/javase/8/docs/api/java/lang/Throwable.html#fillInStackTrace--). The format of this information depends on the implementation, but the following example may be regarded as typical:

Reply:
Oups, sorry